docs/processes/qualityassurance/git-gontribution-guidelines.md

```diff
@@ -0,0 +1,224 @@
+- **test** : Adding missing tests or correcting existing tests
+
+Doubtfire-LMS recommends Chris Beam's post on [How to Write Good Commit Messages](http://chris.beams.io/posts/git-commit/) to improve your commit message writing.
+
+**Use the imperative mood in your commit subject line**
+
+Write your commits in the imperative mood and not the indicative mood
+
+- "Fix a bug" and **not** "Fix*ed* a bug"
+- "Change the behaviour of Y" and **not** "_Changed_ the behaviour of Y"
+- "Add new API methods" and **not** "Sweet new API methods"
+
+Keep the subject line (top line) concise; keep it **within 50 characters**.
+
+Use the body (lines after the top line) to explain why and what and _not_ how; keep it **within 72 characters**.
+
+### But how can I write new lines if I'm using `git commit -m "Message"`?
+
+Don't use the `-m` switch. Use a text editor to write your commit message instead.
+
+If you are using the command line to write your commits, it is useful to set your git editor to make writing a commit body easier. You can use the following command to set your editor to Visual Studio Code, `nano`, `emacs`, `vim`, `atom`.
+
+```bash
+git config --global core.editor "code --wait"
+git config --global core.editor nano
+git config --global core.editor emacs
+git config --global core.editor vim
+git config --global core.editor "atom --wait"
+```
+
+# Git Workflow Summary
+
+## Start a new piece of work
+
+1. Synch repo and set up for new feature branch (remember to use Thoth Tech [branching guidelines](#__Branching_Guide) for naming the new branch):
+
+```bash
+git checkout main
+git pull main
+git checkout -b feature/\<project\_subtask\_description\>
+```
+
+2. Make changes, commit (using comments format that follows [commit guidelines](#_Commit_guidelines)) and push to origin:
+
+```bash
+git add .
+git commit
+git push -u origin feature/\<project\_subtask\_description\>
+```
+
+3. Remember to submit a [draft pull request](#draft-pull-request) via GitHub to allow for code review (and mark as ready to submit when ready to merge your changes to main).
+
+## Continue a piece of work
+
+1. If the branch has already been created in the remote repository.
+
+```bash
+git checkout feature/\<project\_subtask\_description\>
+```
+
+2. Pull in any new code from the default branch
+
+```bash
+git pull origin main
+```
+
+3. Resolve any merge conflicts that may now be revealed.
+
+4. Continue to Step 2 in [Start a new piece of work](#start-a-new-piece-of-work) flow
+
+## Collaborate on an existing branch
+
+Ideally, we should avoid having developers working on the same code. It creates merge conflicts and hinders efficiency. If possible, try to break it down into small tasks so developers can work independently. In the worst-case scenario, if there is more than 1 person working on a feature branch:
+
+1. Please make sure you pull changes in the remove branch before starting your work.
+
+git pull origin feature/\<project_subtask_description\>
```

Comment:
**suggestion:** Similar to the suggestion above ☝🏼, could we use an example branch name to avoid escaping?